OpenRA.Game/Graphics/SpriteRenderer.cs

```diff
@@ -69,41 +69,57 @@ int2 SetRenderStateForSprite(Sprite s)
 
 			// Check if the sheet (or secondary data sheet) have already been mapped
 			var sheet = s.Sheet;
-			var sheetIndex = 0;
-			for (; sheetIndex < ns; sheetIndex++)
-				if (sheets[sheetIndex] == sheet)
+			var sheetIndex = -1;
+			var addCount = 2;
```

Comment:
Shouldn't this be 1 by default and only increased by one if we have secondary data?

Reply:
I believe that was the bug. When both sheet and secondary sheet did not exist - only one new sheet was added both pointing to that same new sheet.

Reply:
Was this an existing bug, or a mistake in this PR?

If it was an existing bug then this will probably fix https://github.com/OpenRA/ra2/issues/710.

Reply:
That was the existing bug that this fix tries to solve.